Backend/Core/Services/WareService.cs

```diff
@@ -83,5 +84,56 @@ public async Task CreateAsync(CreateWareDTO createWareDTO, string userId)
 
             await _characteristicRepository.AddRangeAsync(characteristics);
         }
+
+        public async Task<PaginatedList<WareInfoDTO>> GetByCategoryAsync(
+            PaginationFilterDTO paginationFilter,
+            string categoryTitle)
+        {
+            if (!await _categoryRepository.AnyAsync(
+               new CategorySpecification.GetByTitle(categoryTitle)))
+            {
+                throw new HttpException(
+               ErrorMessages.CategoryNotFound,
+               HttpStatusCode.NotFound);
+            }
+
+            var waresCount = await _wareRepository.CountAsync(
+                new WareSpecification.GetByCategory(paginationFilter, categoryTitle));
+
+            int totalPages = PaginatedList<WareInfoDTO>
+                .GetTotalPages(paginationFilter, waresCount);
+
+            if (totalPages == 0)
+            {
+                return null;
+            }
+
+            var wares = await _wareRepository.ListAsync(
+                new WareSpecification.GetByCategory(paginationFilter, categoryTitle));
+
+            var result = new List<WareInfoDTO>();
+
+            foreach (var ware in wares)
+            {
+
+                result.Add(new WareInfoDTO
+                {
+                    Title = ware.Title,
+                    Description = ware.Description,
+                    Cost = ware.Cost,
+                    PhotoLink = ware.PhotoLink,
```

Comment:
What will you do with the filename on the frontend?
An array of bytes is required to display the photo